### ISSUE TITLE
Create AOI from a shapefile tool cannot handle shapefiles with space in the file name

### ISSUE THREAD

**Issue by jdduh** [PSU-CSAR/vb-bagis#30]
Please automatically replace spaces with underbars.

**Comment by lbross** [comment]
I didn't know ArcMap would let you create shapefile names with spaces in them. I will take a look.

**Comment by lbross** [comment]
We need to think this through some. I can't just replace spaces with underbars because the files on disk have a space in the name. As you know, shapefiles are composed of several files. I could try to rename the shapefile before starting using ArcObjects, but that would fail if the user has the file open in ArcMap.

I could also check for spaces and tell the user they need to rename the shapefile without spaces before continuing and exit gracefully.

Also, should we be concerned if the user puts a space in the output AOI name? That is easy to detect and replace with an underbar before starting processing.

**Comment by jdduh** [comment]
Here is the error message. I'm not sure if it's the space in the shapefile name or in the temp GRID file that caused the problem. If it's the temporary file name, then replacing space with underbar might fix the problem. If the error is caused by the shapefile name, then prompt users and exit gracefully.

![image](https://cloud.githubusercontent.com/assets/11510708/23324340/09b928bc-faa2-11e6-8df6-09a71c65ba2a.png)

**Comment by jdduh** [comment]
It seems BAGIS/-H/-P can handle AOI names that contain spaces. We can allow the output AOI name to have space in it if the space doesn't affect the creation of AOI.

**Comment by lbross** [comment]
1. It seems that the RasterConversionOp in BA_ShapeFile2RasterGDB cannot handle a space in the shapefile name. If it's important that we support spaces in the shapefile name, I could try to swap in the GP Feature to Raster tool. Or we can require the user to supply a shapefile without a space in the name.

2. BAGIS-H doesn't support AOIs with a space in the name. I created an AOI with a space in the name in BAGIS and then couldn't open it in BAGIS-H. I have implemented validation to replace spaces with underbars in the new AOI name. 

3. I found another bug in this tool preventing it from clipping the pourpoint layer. I have fixed this and it will be in the next version of BAGIS.

So ... let me know thoughts on item #1 to wrap this up.
